crates/biome_css_parser/src/syntax/selector.rs

```diff
@@ -1,14 +1,46 @@
+use crate::lexer::CssLexContext;
 use crate::parser::CssParser;
-use crate::syntax::parse_error::expected_identifier;
+use crate::syntax::parse_error::{expect_any_selector, expected_identifier};
 use crate::syntax::{is_at_identifier, parse_error, parse_identifier};
 use biome_css_syntax::CssSyntaxKind::*;
 use biome_css_syntax::{CssSyntaxKind, T};
 use biome_parser::parse_lists::ParseNodeList;
 use biome_parser::parse_recovery::{ParseRecovery, RecoveryResult};
 use biome_parser::prelude::ParsedSyntax;
 use biome_parser::prelude::ParsedSyntax::{Absent, Present};
-use biome_parser::{token_set, Parser, TokenSet};
+use biome_parser::{token_set, CompletedMarker, Parser, ParserProgress, TokenSet};
 
+#[inline]
+pub(crate) fn parse_selector(p: &mut CssParser) -> ParsedSyntax {
+    parse_compound_selector(p).and_then(|selector| try_parse_complex_selector(p, selector))
+}
+
+#[inline]
+pub(crate) fn try_parse_complex_selector(p: &mut CssParser, left: CompletedMarker) -> ParsedSyntax {
```

Comment:
In Rust, functions that start with `try_*` tend to return errors with `Result`, but here we don't. We should at least document what this function **tries** to do and how.

Reply:
renamed to `parse_complex_selector`.